### PR TITLE
Replace tqdm progress bars with rich

### PR DESCRIPTION
**Pull request by eulertour**
Replaces the progress bars with those from rich.

### CODE REVIEW

**Comment by kilacoda** [comment]
Sorry for being nitpicky, but can the time elapsed also be added as a column?

File: manim/scene/scene.py [hunk 1]
```diff
@@ -652,91 +653,105 @@ def get_moving_mobjects(self, *animations):
...
+        Shortens a description to a maximum length of
+        self.max_description_width if necessary and pads it with ... if it was
+        shortened.
 
         Parameters
         ----------
-        animations : list of Animation
-            The list of animations to get
-            the time progression for.
+        description: str
```
Comment (PgBiel) [inline manim/scene/scene.py:742]: `` :class:`str` ``  (so sphinx renders it correctly)
```suggestion
        description: :class:`str`
```

File: manim/scene/scene.py [hunk 2]
```diff
@@ -652,91 +653,105 @@ def get_moving_mobjects(self, *animations):
...
-        run_time = self.get_run_time(animations)
-        time_progression = self.get_time_progression(run_time)
-        time_progression.set_description("".join([
-            "Animation {}: ".format(self.num_plays),
-            str(animations[0]),
-            (", etc." if len(animations) > 1 else ""),
-        ]))
-        return time_progression
+        -------
+        str
```
Comment (PgBiel) [inline manim/scene/scene.py:747]:
```suggestion
        :class:`str`
```

File: manim/scene/scene.py [hunk 3]
```diff
@@ -652,91 +653,105 @@ def get_moving_mobjects(self, *animations):
...
+    def formatted_progress(self):
+        """
+        Returns a GlobalProgress formatted to display the number of elapsed
+        frames.
 
         Returns
         ------
-        ProgressDisplay
-            The CommandLine Progress Bar.
+        GlobalProgress
```
Comment (PgBiel) [inline manim/scene/scene.py:681]: For correct sphinx rendering
```suggestion
        :class:`GlobalProgress`
```

File: manim/scene/scene.py [hunk 4]
```diff
@@ -652,91 +653,105 @@ def get_moving_mobjects(self, *animations):
...
         ----------
-        run_time: float
-            The run_time of the animation.
+        animations: list of Animation
+            A list of the animations whose total
+            run_time is to be calculated.
+
+        Returns
+        ------
+        float
```
Comment (PgBiel) [inline manim/scene/scene.py:668]: For correct sphinx rendering
```suggestion
        :class:`float`
```

File: manim/scene/scene.py [hunk 5]
```diff
@@ -652,91 +653,105 @@ def get_moving_mobjects(self, *animations):
...
-        the iterations to perform in that animation
-        and a bool saying whether or not to consider
-        the skipped animations.
+        Gets the total run time for a list of animations.
 
         Parameters
         ----------
-        run_time: float
-            The run_time of the animation.
+        animations: list of Animation
```
Comment (PgBiel) [inline manim/scene/scene.py:662]: For consistency etc.
```suggestion
        animations: List[:class:`~.Animation`]
```

**Comment by eulertour** [comment]
Closing as the rich progress bars are too slow.

**Pull request closed without merging**